src/methods.jl

```diff
@@ -77,3 +74,54 @@ julia> subtract_overscan(frame, (:, 4:5), dims = 2)
 * [`subtract_overscan!`](@ref)
 """
 subtract_overscan(frame::AbstractArray, idxs; dims = axes_min_length(idxs)) = subtract_overscan!(deepcopy(frame), idxs, dims = dims)
+
+
+"""
+    flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; norm_value = mean(flat_frame))
+
+In-place version of [`flat_correct`](@ref)
+"""
+function flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; norm_value = mean(flat_frame))
+    size(frame) != size(flat_frame) && error("size of frame and flat frame are not same")
```

Comment:
So, I'm curious about needing this at all. This check is done automatically when we do the broadcasting, which also mentions the mismatched dimensions and size, so adding it here isn't really necessary other than saying `frame` and `flat frame` instead of `array A` and `array B`. Thought @giordano ?

Reply:
Yes, when using broadcasting this is probably redundant. It's necessary when indexing directly, like what is done in `subtract_bias!`

Reply:
Then let's switch to broadcasting in `subtract_bias!` as well, it maintains the uniformity in the code and also broadcast is GPU compatible. What would you suggest? @giordano @mileslucas

Reply:
Sure, easy enough